### PR TITLE
docs: Update data security

### PR DESCRIPTION
**Pull request by bmorelli25**
## Summary

This PR updates the APM Data Security documentation. Click below to preview these changes:

* [Data security](https://apm-server_8043.docs-preview.app.elstc.co/guide/en/apm/guide/master/apm-data-security.html)
  * [Built-in data filters](https://apm-server_8043.docs-preview.app.elstc.co/guide/en/apm/guide/master/filtering.html)
  * [Custom filters](https://apm-server_8043.docs-preview.app.elstc.co/guide/en/apm/guide/master/custom-filter.html)
  * [Delete sensitive data](https://apm-server_8043.docs-preview.app.elstc.co/guide/en/apm/guide/master/data-security-delete.html)

## Related issues

* Closes https://github.com/elastic/apm-server/issues/5087.
* Closes https://github.com/elastic/observability-docs/issues/1228.
* Related to https://github.com/elastic/observability-docs/issues/1226.

### CODE REVIEW

File: docs/apm-data-security.asciidoc [hunk 1]
```diff
@@ -18,26 +32,54 @@ Elastic APM provides built-in support for filtering the following types of data:
...
+| `http.request.body.original` | The body of the monitored HTTP request. | <<filters-http-body>>
+| `http.request.headers` | The canonical headers of the monitored HTTP request. | <<filters-http-header>>
+| `http.request.socket.remote_address` | The address of the last proxy or end-user (if no proxy). | <<custom-filter>>
+| `http.response.headers` | The canonical headers of the monitored HTTP response. | <<filters-http-header>>
+| `process.args` | Process arguments. | <<filters-database-statements>>
+| `span.db.statement` | Database statement. | <<filters-database-statements>>
+| `stacktrace.vars` | A flat mapping of local variables captured in the stack frame | <<custom-filter>>
+| `url.query` | The query string of the request, e.g. `?pass=hunter2`. | <<custom-filter>>
+| `user.*` | Logged-in user information. | <<custom-filter>>
+| `user_agent.*` | Device and version making the network request. | <<filters-personal-data>>
```
Comment (bmorelli25) [inline docs/apm-data-security.asciidoc:74]: What other fields should users review?

File: docs/apm-data-security.asciidoc [hunk 2]
```diff
@@ -152,31 +194,32 @@ Certain agents offer additional filtering and obfuscating options:
...
 [[filters-ingest-pipeline]]
 ==== Create an ingest node pipeline filter
 
 Ingest node pipelines specify a series of processors that transform data in a specific way.
-Transformation happens prior to indexing–inflicting no performance overhead on the monitored application.
+Transformation happens prior to indexing--inflicting no performance overhead on the monitored application.
 Pipelines are a flexible and easy way to filter or obfuscate Elastic APM data.
 
+WARNING: Changes to the default APM pipeline do not persist through upgrades.
+You will need to re-add custom ingest pipelines after every upgrade.
```
Comment (bmorelli25) [inline docs/apm-data-security.asciidoc:218]: How do we feel about documenting this workaround? I know in the past we said we didn't want to do this (https://github.com/elastic/apm-server/pull/6940#discussion_r782735208), but I don't see another option.

File: docs/apm-data-security.asciidoc [hunk 3]
```diff
@@ -455,4 +483,97 @@ Please see the relevant agent's documentation for more information and examples:
...
+. Delete or redact the ingested data. With data collection fixed, you can now delete or redact the offending data:
++
+* <<redact-field-data>>
+* <<delete-doc-data>>
+
+[float]
+[[redact-field-data]]
+===== Redact specific fields
+
+To redact sensitive data in a specific field, use the {ref}/docs-update-by-query.html[update by query API].
```
Comment (bmorelli25) [inline docs/apm-data-security.asciidoc:508]: Is this the best way? What about the two examples below? Are there better ones that you can think of?

File: docs/apm-data-security.asciidoc [hunk 4]
```diff
@@ -455,4 +483,97 @@ Please see the relevant agent's documentation for more information and examples:
...
+    }
+  }
+}
+----
+
+See {ref}/docs-update-by-query.html[update by query API] for more information and examples.
+
+[float]
+[[delete-doc-data]]
+===== Delete {es} documents
```
Comment (bmorelli25) [inline docs/apm-data-security.asciidoc:557]: Do you see a use case where someone would want to delete entire documents?

**Review by bmorelli25 — COMMENTED** [review]
Here's a draft of the updated data security content. I have a few questions for ya, @elastic/apm-server.

File: docs/apm-data-security.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/apm-data-security.asciidoc:218]: Hmmm. It looks like config changes also overwrite the pipeline. I'm not sure how to move forward here.

**Comment by bmorelli25** [comment]
I'm still looking for feedback on this one, @elastic/apm-server 🙇

File: docs/apm-data-security.asciidoc (hunk 2 above)
Comment (simitt) [reply docs/apm-data-security.asciidoc:218]: I don't see any other option than documenting that custom ingest pipelines are not supported in the integration at this point. There are plans to add this functionality to the Fleet eco system, but until it lands there's nothing much we can do. 

> Hmmm. It looks like config changes also overwrite the pipeline.

@joshdover is this on purpose?

File: docs/apm-data-security.asciidoc (hunk 4 above)
Comment (simitt) [reply docs/apm-data-security.asciidoc:557]: Nothing immediately comes to my mind, but I don't see harm in having the examples.

File: docs/apm-data-security.asciidoc (hunk 3 above)
Comment (simitt) [reply docs/apm-data-security.asciidoc:508]: Examples LGTM

File: docs/apm-data-security.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/apm-data-security.asciidoc:218]: Okey dokey. I'll comment out all of the ingest pipeline examples for now.

File: docs/apm-data-security.asciidoc (hunk 3 above)
Comment (bmorelli25) [reply docs/apm-data-security.asciidoc:508]: Thanks 😃

File: docs/apm-data-security.asciidoc (hunk 4 above)
Comment (bmorelli25) [reply docs/apm-data-security.asciidoc:557]: 👍 Cool beans

File: docs/apm-data-security.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/apm-data-security.asciidoc]: All ingest node content removed in [`5b24889`](https://github.com/elastic/apm-server/pull/8043/commits/5b24889e7f15a5c8f9e2dfc16dce7c960272dab5).

**Comment by bmorelli25** [comment]
Ingest pipeline content has been commented out. When we have the ability to add custom pipelines to integrations, I'll update these docs and others. Tracking in https://github.com/elastic/observability-docs/issues/1226.

**Pull request merged**